### PR TITLE
Add more options to filter articles in API

### PR DESCRIPTION
**Pull request by ahilles107**
| Q                         | A
| ------------------------- | ---
| Bug fix?                  | no
| New feature?              | yes
| BC breaks?                | no
| Deprecations?             | no
| Tests pass?               | yes
| Changelog update required | yes
| Fixed tickets             | SWP-660
| License                   | AGPLv3

### CODE REVIEW

File: src/SWP/Bundle/ContentBundle/Controller/ArticleController.php [hunk 1]
```diff
@@ -55,10 +60,19 @@ class ArticleController extends Controller
...
     {
+        $authors = '';
+        if (null !== $request->query->get('author', null)) {
+            $authors = explode(', ', $request->query->get('author'));
+        }
+
         $articles = $this->get('swp.repository.article')
             ->getPaginatedByCriteria(new Criteria([
-                'status' => $request->query->get('status', ''),
+                'status' => $request->query->get('status', ArticleInterface::STATUS_PUBLISHED),
```
Comment (takeit) [inline src/SWP/Bundle/ContentBundle/Controller/ArticleController.php]: so by default it should list only published articles? not all?

Comment (ahilles107) [inline src/SWP/Bundle/ContentBundle/Controller/ArticleController.php]: Hmm... from other side - it's internal API. I woud expect to see only bublished allways, but then we would need add option to get all of them. So ok, i will revert this one.

File: src/SWP/Bundle/ContentBundle/Controller/ArticleController.php [hunk 2]
```diff
@@ -55,10 +60,32 @@ class ArticleController extends Controller
...
+            $routeObject = $this->get('swp.provider.route')->getOneById($request->query->get('route'));
+
+            if (null !== $routeObject) {
+                $ids = [$routeObject->getId()];
+                foreach ($routeObject->getChildren() as $child) {
+                    $ids[] = $child->getId();
+                }
+            }
+
+            $request->query->set('route', $ids);
```
Comment (takeit) [inline src/SWP/Bundle/ContentBundle/Controller/ArticleController.php]: https://scrutinizer-ci.com/g/superdesk/web-publisher/inspections/2bb434c8-5659-472d-a037-c3c5fd7ff6e5/issues/files/src/SWP/Bundle/ContentBundle/Controller/ArticleController.php?status=new&orderField=path&order=asc&honorSelectedPaths=0

Comment (ahilles107) [inline src/SWP/Bundle/ContentBundle/Controller/ArticleController.php]: i see

**Pull request merged**